src/biogeochem/CNPhenologyMod.F90

```diff
@@ -972,6 +1034,147 @@ subroutine CNSeasonDecidPhenology (num_soilp, filter_soilp       , &
+                       result( do_onset )
+
+    !
+    ! !DESCRIPTION:
+    ! Function to determine if seasonal decicious leaf onset should happen.
+    !
+    ! !USES:
+    use shr_const_mod   , only: SHR_CONST_TKFRZ
+    ! !ARGUMENTS:
+    real(r8), intent(INOUT) :: onset_gdd      ! onset growing degree days 
+    real(r8), intent(INOUT) :: onset_gddflag  ! Onset freeze flag
+    real(r8), intent(IN)    :: soilt          ! Soil temperture at specific level for this evaluation
+    real(r8), intent(IN)    :: soila10        ! 10-day running mean of the 12cm soil layer temperature (K)
+    real(r8), intent(IN)    :: t_a5min        ! 5-day running mean of min 2-m temperature
+    real(r8), intent(IN)    :: dayl           ! Day length
+    real(r8), intent(IN)    :: snow_5day      ! 5-day average of snow
+    real(r8), intent(IN)    :: ws_flag        ! winter-summer solstice flag (0 or 1)
+    real(r8), intent(IN)    :: crit_onset_gdd ! critical onset growing degree-day sum
+    real(r8), intent(IN)    :: season_decid_temperate  ! If this is a temperate seasonal decidious type 
+    logical :: do_onset                       ! Flag if onset should happen (return value)
+    !
+    ! !LOCAL VARIABLES:
+    real(r8), parameter :: snow5d_thresh_for_onset        = 0.1_r8          ! 5-day snow depth threshold for onset
+    real(r8), parameter :: min_critial_daylength_onset    = 39300._r8/2._r8 !  Minimum daylength for onset to happen
+    !-----------------------------------------------------------------------
+
+    do_onset = .false.
+    ! Test to turn on growing degree-day sum, if off.
+    ! switch on the growing degree day sum on the winter solstice
+
+    if (onset_gddflag == 0._r8 .and. ws_flag == 1._r8) then
+        onset_gddflag = 1._r8
+        onset_gdd = 0._r8
+    end if
+
+    ! Test to turn off growing degree-day sum, if on.
+    ! This test resets the growing degree day sum if it gets past
+    ! the summer solstice without reaching the threshold value.
+    ! In that case, it will take until the next winter solstice
+    ! before the growing degree-day summation starts again.
+
+    if (onset_gddflag == 1._r8 .and. ws_flag == 0._r8) then
+        onset_gddflag = 0._r8
+        onset_gdd = 0._r8
+    end if
+
+    ! if the gdd flag is set, and if the soil is above freezing
+    ! then accumulate growing degree days for onset trigger
+
+    if (onset_gddflag == 1.0_r8 .and. soilt > SHR_CONST_TKFRZ) then
+        onset_gdd = onset_gdd + (soilt-SHR_CONST_TKFRZ)*fracday
+    end if
+    if ( onset_thresh_depends_on_veg) then
+       ! separate into non-arctic seasonally deciduous pfts
+       ! (temperate broadleaf deciduous
+       ! tree) and arctic/boreal seasonally deciduous pfts (boreal
+       ! needleleaf deciduous tree,
+       ! boreal broadleaf deciduous tree, boreal broadleaf deciduous
+       ! shrub, C3 arctic grass)
+       if (onset_gdd > crit_onset_gdd .and.  season_decid_temperate == 1) then
+           do_onset = .true.
```

Comment:
The check "dayl>min_critical_daylength_onset" in the if statement was added because...."

Reply:
Got it